### ISSUE TITLE
stuck at "INFO:list for subsample is not initialized. Initializing now" for about a day

### ISSUE THREAD

**Issue by gimnec** [yfukasawa/LongQC#16]
info:
SRR10520596.fastq is ~3 GB in size

command:
python longQC.py sampleqc -x pb-rs2 -o /data/out_dir /data/SRR10520596.fastq

logs:
longQC:2020-12-30 16:26:53,142:169:INFO:Cmd: longQC.py sampleqc -x pb-rs2 -o /data/out_dir /data/SRR10520596.fastq
longQC:2020-12-30 16:26:53,143:233:INFO:Preset "pb-rs2" was applied. Options --pb(--ont) is overwritten.
longQC:2020-12-30 16:26:58,169:306:INFO:Computation of the low complexity region started for a chunk 0
lq_mask:2020-12-30 16:27:13,623:111:INFO:New job was submitted: in->/data/out_dir/analysis/tmp_0.fastq, out->/data/out_dir/analysis/tmp_0.txt
longQC:2020-12-30 16:27:13,630:311:INFO:Adapter search is starting for a chunk 0.
longQC:2020-12-30 16:27:13,635:327:INFO:Computation of the GC fraction started for a chunk 0
lq_utils:2020-12-30 16:27:21,706:380:INFO:list for subsample is not initialized. Initializing now.

It freezed here for a day that I had to force quit.

**Comment by yfukasawa** [comment]
Hello @gimnec,

I'll try to reproduce the error with that SRA fastq.
At present, the problem looks to be related to your file system.

Are you using some computer cluster?
If so, could you try again to see if the same issue happens?

Yoshinori

**Comment by gimnec** [comment]
@yfukasawa 

I was not using any computer cluster, just a Mac.
I'm now running on another Windows 10 pc, and it went further than on Mac. But it has been freezing at "longQC:2020-12-31 13:54:53,592:327:INFO:Computation of the GC fraction started for a chunk 1" for an hour now.

Prior to that, I've modified the original Dockerfile because it gave me errors that the pysam and edlib versions are not compatible with others. Here's the modified version.
-------------------------------------------------------------------------------
# build minimap2-coverage
FROM continuumio/miniconda3
### MAINTAINER ###
MAINTAINER Yoshinori Fukasawa <yoshinori.fukasawa@kaust.edu.sa>

RUN apt-get clean all && \
    apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y  \
    git \
    build-essential \
    libc6-dev \
    zlib1g-dev && \
    apt-get clean && \
    apt-get purge

### ENV VARS ###
ENV USER user
ENV HOME /home/${USER}

### LABELS ###
LABEL base_image="miniconda3"
LABEL software="LongQC docker"
LABEL software.version="1.2"

# add a general user account
RUN useradd -m ${USER}
# define a password for user
RUN echo "${USER}:test_pass" | chpasswd

ADD https://api.github.com/repos/yfukasawa/longqc/git/refs/heads/minimap2_update version.json
RUN git clone https://github.com/yfukasawa/LongQC.git $HOME/LongQC
RUN cd $HOME/LongQC/minimap2-coverage && make

# install dependency
RUN conda update -y conda

RUN conda config --add channels defaults
RUN conda config --add channels bioconda
RUN conda config --add channels conda-forge
RUN conda config --add channels anaconda
RUN conda config --add channels r

RUN conda install -y numpy
RUN conda install -y pandas'>=0.24.0'
RUN conda install -y scipy
RUN conda install -y jinja2
RUN conda install -y h5py
RUN conda install -y matplotlib'>=2.1.2'
RUN conda install -y scikit-learn

RUN pip install pysam
RUN pip install edlib

# RUN conda install -y -c bioconda pysam
# RUN conda install -y -c bioconda edlib
RUN conda install -y -c bioconda python-edlib

# change user to "user" defined above
USER ${USER}

# define a working dir
WORKDIR $HOME
-------------------------------------------------------------------------------

**Comment by yfukasawa** [comment]
Thanks.

Then, adapter seq alignment using edlib might be causing issues in your env.
I'll try to reproduce the issue with the same dataset.

In the meantime, may I ask versions for pysam, edlib, and python-edlib?

Best,
Yoshinori

**Comment by gimnec** [comment]
@yfukasawa 

Thank you in advance.

Those versions are:
pysam 0.16.0.1
edlib 1.3.8.post2
python-edlib 1.3.8.post2
![Clip_24](https://user-images.githubusercontent.com/76786322/103479049-569b1200-4dfd-11eb-825c-2faf879e050d.png)

**Comment by yfukasawa** [comment]
Hi @gimnec,

I've tried to build Docker file ([Dockerfile.txt](https://github.com/yfukasawa/LongQC/files/5761745/Dockerfile.txt)) with your modifications after removing all of containers, images and caches on my Mac.

1. I still can't reproduce pysam/edlib incompatible errors (installation through conda worked ok), and the conflicting might have been caused by pre-existing Docker image/caches in your environment. If you try to cleanly install, the error may disappear. otherwise, kindly let me know.
2. python-edlib version 1.3.8.post2 worked fine in my env.

To pinpoint the issue, could you try test the adapter module? If this is ok, I can narrow down issues. 
For this, I found some minor bugs in the code, and fixed and commit the patch just before.

In Docker, please clone the latest repo.
then, please change [the path for the input in the file](https://github.com/yfukasawa/LongQC/blob/master/lq_adapt.py#L107).
In my case, the file was put under /data/.
Then run just this module as follow, and it should finish quickly (~ a few mins).

```
time python LongQC/lq_adapt.py
lq_adapt:2021-01-03 14:36:09,659:119:INFO:Input file: /data/SRR10520596.fastq
lq_adapt:2021-01-03 14:37:14,474:121:INFO:File was loaded.
lq_adapt:2021-01-03 14:37:25,144:77:INFO:9074 reads were skipped due to their short lengths.
lq_adapt:2021-01-03 14:37:25,145:90:INFO:Adapter Sequence: ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT, max identity:1.000000 and the number of trimmed reads: 502
lq_adapt:2021-01-03 14:37:37,344:42:INFO:9074 reads were skipped due to their short lengths.
lq_adapt:2021-01-03 14:37:37,345:92:INFO:Adapter Sequence: ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT, max identity:1.000000 and the number of trimmed reads: 1290

real	1m36.976s
user	0m55.044s
sys	0m14.878s
```

If this doesn't finish, python-edlib or something related to this behaves oddly in your env. if works fine, kindly let me know.
Yoshinori

**Comment by gimnec** [comment]
@yfukasawa 

I think I solved this problem now. By trying many ways from the start, I eventually found out that there are 2 main errors solved by modifying Dockerfile as follows.
![Clip_25](https://user-images.githubusercontent.com/76786322/103746440-af53f000-5033-11eb-8dbc-68fac9334185.png)
![Clip_26](https://user-images.githubusercontent.com/76786322/103746513-cabefb00-5033-11eb-9e65-d7cf293ddbc2.png)

Anyway, I'm coming up with the new issues. I'll post it in another thread.

And thank you so much for your help!

**Comment by gimnec** [comment]
On my PC, I modified Dockerfile like the above comment, then built and ran it, and it went farther than "INFO:list for subsample is not initialized. Initializing now", although did not finish because of another error.
I then saved a docker image of LongQC, and loaded it on Mac at my working place. It was stuck at "INFO:list for subsample is not initialized. Initializing now", unlike on my PC that it went further this point.
I think the root cause of this strange problem could be the strict proxy of my working place, so I'll have to continue working on my PC for now.
